### DISCUSSION TITLE
How can you tell what version of Roslyn is included with an SDK?

### DISCUSSION THREAD

**Discussion by parched** [dotnet/roslyn#75031] [Q&A] [answered]
Specifically, I would like to know what version has [this fix](https://github.com/dotnet/roslyn/pull/73339/commits/fa907c7ba9a834342e3b53fac04defd3707699b1). I see a bunch of tags in this repo, but they mention VS or VS Code versions, what if I'm using the command line?

**Comment by huoyaoyuan** [comment]
The minor versions of VS and Roslyn, and feature bands of .NET SDK are currently tightened. For example:
|Visual Studio|Roslyn|.NET SDK|
|-|-|-|
|17.12 (Future)|4.12|9.0.1xx|
|17.11|4.11|8.0.4xx|
|17.10|4.10|8.0.3xx|

At the download page of .NET SDK (https://dotnet.microsoft.com/download/dotnet/8.0), you can use *Included in* to decide relationship between SDK version and VS version. You can also inspect the version config from SDK repo: https://github.com/dotnet/sdk/blob/3f8227e12bb5b84ba9a339b04e0d48185434947e/eng/Versions.props#L161-L166

**Reply by parched** [reply]
Thanks. How can you go from those versions to a commit in this repo?

**Reply by 333fred** [reply]
The easiest way to determine this, given an SDK, is simply to create a quick test project, put `#error version` in a `.cs` file, and let the compiler tell you what version and commit it is.

**Reply by parched** [reply]
Thanks, good to know. Is there any way without downloading every version I want to check?

**Reply by 333fred** [reply]
Not afaik.

**Reply by brckelly** [reply]
Is there a version of this table that is kept up-to-date?

**Comment by JoeRobich** [comment, marked as answer]
@parched I have pushed tags for SDK releases. Should be easier to find now.

**Comment by WeihanLi** [comment]
Hope it could be included into the `dotnet --info` output, created an issue for this before
https://github.com/dotnet/sdk/issues/44041
